src/modules/roc_core/target_posix/roc_core/colors.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "roc_core/colors.h"
+
+namespace roc {
+namespace core {
+
+//! ANSI Color Codes.
+#define COLOR_NONE       ""
+#define COLOR_START      "\033["
+#define COLOR_END        "m"
+#define COLOR_RESET      COLOR_START COLOR_END
+#define COLOR_SEPARATOR  ";"
+#define COLOR_BOLD       "1"
+#define COLOR_RED        "31"
+#define COLOR_BLUE       "34"
+
+
+static char const* colors_to_levels(LogLevel level) {
+    char const* color = COLOR_NONE;
+    switch (level) {
+        case LogNone:   //!< No color for Loglevel None.
+            break;
+        case LogError:  //!< Bold Red for Loglevel Error.
+            color = COLOR_START COLOR_BOLD COLOR_SEPARATOR \
+                    COLOR_RED COLOR_END;
+            break;
+        case LogInfo:   //!< Bold Blue for Loglevel None.
+            color = COLOR_START COLOR_BOLD COLOR_SEPARATOR \
+                    COLOR_BLUE COLOR_END;
+            break;
+        case LogDebug:  //!< No color for Loglevel None.
+            break;
+        case LogTrace:  //!< No color for Loglevel None.
+            break;
+    }
+    return color;
+}
+
+bool colors_available() {
+    char * term = getenv("TERM");
+    return isatty(STDERR_FILENO) && \
+           term && strncmp("dumb", term, 4) != 0;
+}
+
+bool format_colored(LogLevel level, const char* str,
+                    char* buf, size_t bufsz) {
```

Comment:
Lets add some constraints to the function:

```cpp
roc_panic_if_not(str);
roc_panic_if_not(buf);
```

to ensure we won't receive UB.